fidesctl/src/fidesctl/core/evaluate.py

```diff
@@ -500,12 +501,35 @@ def populate_referenced_keys(
     return taxonomy
 
 
+def populate_evaluation_policies(
```

Comment:
Two things here:
1. It's not a big deal right now, but I think there's a dangerous precedent in play. We don't want to get stuck having to pass `local` to every function call just to enable the evaluation of conditionals like this. Is there a way we can store the `local` boolean value at a high enough scope (ideally not global) that it can be readily accessed?
2. I'd like it if the fetching and validation concerns could be separated. It would mean decomposing this function into two helpers. This is valuable as a general best practice, but also:
	- The fetch helper wouldn't need to receive `local`, since it would only be called by `evaluate` when not in local mode (eliminating one level of having to pass this value down)
	- The validation helper shouldn't _care_ about the value of `local` at all

Reply:
I think the `evaluate` function is that higher level. Coming from a functional point of view, I think it's more dangerous to _not_ pass in every parameter specifically. I think the right level of abstraction is to move the if statement up one level into the `evaluate` function. That should satisfy both of the asks here

Reply:
Yes - that's more or less what I had in mind. The `evaluate` function would contain a conditional to guard `fetch`, and then run `validate` once the correct policies have been collected for validation. Then, we don't need to pass `local` any further down than `evaluate` itself.

Reply:
change made, good call 👍 

i actually initially moved that stuff into its own function because `evaluate` was tripping `xenon`, but your comment made me realize how silly of a move it was to increase complexity to satisfy a static analysis tool, so I just loosened xenon's requirements slightly and restored it to how it was before with the boolean check being done in `evaluate`